### PR TITLE
Add op to validate ref and remote names.

### PR DESCRIPTION
**Pull request by jdgarrett**

### CODE REVIEW

File: src/core/src/main/java/org/locationtech/geogig/api/plumbing/CheckRefFormat.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+import org.locationtech.geogig.api.AbstractGeoGigOp;
+import org.locationtech.geogig.api.NodeRef;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+
+/**
+ * Validates the format of a ref.
+ *
```
Comment (groldan) [inline src/core/src/main/java/org/locationtech/geogig/api/plumbing/CheckRefFormat.java]: Please explain what the validation rules are

File: src/core/src/main/java/org/locationtech/geogig/api/plumbing/CheckRefFormat.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    private boolean allowOneLevel = true;
+
+    private boolean throwsException = false;
+
+    public CheckRefFormat setRef(String ref) {
+        this.ref = ref;
+        return this;
+    }
+
+    public CheckRefFormat setAllowOneLevel(boolean allow) {
```
Comment (groldan) [inline src/core/src/main/java/org/locationtech/geogig/api/plumbing/CheckRefFormat.java]: Add method javadoc explaining what allowOneLevel does

**Comment by groldan** [comment]
@jdgarrett other than the request to add a couple javadocs, the patch looks alright.

**Comment by jdgarrett** [comment]
Added javadocs to the operation and methods.

**Pull request merged**